src/pkg/caendr/caendr/services/sql/etl/phenotype_metadata.py

```diff
@@ -0,0 +1,70 @@
+import csv
+
+from caendr.services.logger import logger
+
+from caendr.models.datastore  import Species
+from caendr.utils.local_files import LocalDatastoreFile
+
+
+def parse_phenotype_metadata(species: Species, **files: LocalDatastoreFile):
+  """
+    Parsing function to load PhenotypeMetadata table from datastore.
+    For bulk file ouputs each trait as a new row and replicates metadata for all of them
+  """
+  
+  logger.info('Parsing function for traits metadata')
+  
+  for fname, fpath in files.items():
+
+    # Get metadata for the file 
+    md = fpath._metadata
```

Comment:
Quick note on this -- you should be able to get the metadata using `fpath.metadata()`.

Generally in Python, any class method / variable that starts with `_` is considered "private" -- it's a way of signaling to other programmers that you should only access it inside the class definition itself.  Technically in Python, there's no such thing as "private" vs. "public" fields, since the language itself makes all fields public.  However, having "private" fields is often useful to separate implementation from access.

(For example, I might decide later that the internal `_metadata` field needs to change, which could break this code; but, as long as I keep the return value of the external `metadata()` function the same, I can change as much as I want without breaking your code.)

In this case, I've marked the internal `_metadata` field as private for reasons to do with the parent class of `LocalDatastoreFile` -- happy to explain more in-depth if you'd like.

Reply:
Thanks for educating on this! Fixed in the commit 6dff197494274095938bbf64cef3d2accce75ea2